### PR TITLE
Limit access to national identification number and create person

### PR DESCRIPTION
**Pull request by svenngjeto**

### CODE REVIEW

File: cristin-commons/src/main/java/no/unit/nva/utils/AccessUtils.java
Comment (torarnet) [inline cristin-commons/src/main/java/no/unit/nva/utils/AccessUtils.java:15]: Maybe better naming as this validates specific access rights (can se nin), not generic

File: cristin-person/src/main/java/no/unit/nva/cristin/person/client/CristinPersonQuery.java
Comment (torarnet) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/client/CristinPersonQuery.java:23]: Have this just one place and import it elsewhere

File: cristin-person/src/main/java/no/unit/nva/cristin/person/client/AuthorizedCristinPersonApiClient.java
Comment (torarnet) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/client/AuthorizedCristinPersonApiClient.java:22]: Have this just one place and import it elsewhere. Found it also in CristinPersonQuery

File: cristin-person/src/main/java/no/unit/nva/cristin/person/handler/CreateCristinPersonHandler.java
Comment (torarnet) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/handler/CreateCristinPersonHandler.java:51]: This maybe needs other access rights than "Can see national identity number".

File: cristin-person/src/main/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandler.java
Comment (torarnet) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandler.java:28]: Is already present in `ErrorMessages`

File: cristin-person/src/main/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandler.java
Comment (torarnet) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandler.java:84]: Is this now safe after the update?

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandlerTest.java
Comment (torarnet) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandlerTest.java:92]: Is this necessary?

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandlerTest.java
Comment (torarnet) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandlerTest.java:65]: I think this also is in nva-commons

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/personAuthorized.feature
Comment (torarnet) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/handler/personAuthorized.feature:21]: Does this actually return 200? What about authorization?

File: cristin-person/src/main/java/no/unit/nva/cristin/person/client/AuthorizedCristinPersonApiClient.java
Comment (torarnet) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/client/AuthorizedCristinPersonApiClient.java:51]: Also found in `CristinPersonQuery`

File: cristin-person/src/main/java/no/unit/nva/cristin/person/client/AuthorizedCristinPersonApiClient.java
Comment (torarnet) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/client/AuthorizedCristinPersonApiClient.java:34]: This is duplicate code. Have it only one place. Maybe better to move all logic for identity fetch into this authorized client?

File: template.yaml
Comment (torarnet) [inline template.yaml:199]: TODO is Done?

File: template.yaml
Comment (torarnet) [inline template.yaml:29]: Not in use complaints from linter

Comment (svenngjeto) [reply template.yaml:29]: linter doesn't understand the include, it is referenced in docs/cristin-proxy-swagger.yaml

Comment (torarnet) [reply template.yaml:29]: Can you add an lint exclude so we dont get an error?

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandlerTest.java
Comment (svenngjeto) [reply cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandlerTest.java]: Not at all, just to make a valid request at a point

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandlerTest.java
Comment (torarnet) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandlerTest.java:131]: Maybe split this into two tests? I got an error doing exactly the same and had two split into two separate `GatewayResponse` objects instead of reusing the same instance

File: cristin-person/src/main/java/no/unit/nva/cristin/person/client/CristinPersonApiClient.java
Comment (axthosarouris) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/client/CristinPersonApiClient.java:196]: what is upstream?

File: cristin-person/src/main/java/no/unit/nva/cristin/person/client/CristinPersonApiClient.java
Comment (axthosarouris) [inline cristin-person/src/main/java/no/unit/nva/cristin/person/client/CristinPersonApiClient.java:207]: fetchQueryResults -> queryXXXXservice?

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandlerTest.java
Comment (axthosarouris) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandlerTest.java:90]: What is a "correct" Cristin URI?

**Review by axthosarouris — COMMENTED** [review]
This is kind of a bit PR. It is a bit difficult to give a very meaningful review, which will result in overlooking things  and paying double the price by refactoring later

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandlerTest.java
Comment (svenngjeto) [reply cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandlerTest.java]: Renamed test. (It's a path containg cristinId and not NationalIdentificationNumber (N26r))

**Review by axthosarouris — APPROVED** [review]

**Comment by axthosarouris** [comment]
Approved with a small worry that something is escaping my attention because of the PR size. Try to hold the PRs up to 250 lines

File: cristin-person/src/test/java/no/unit/nva/cristin/person/handler/personAuthorized.feature
Comment (torarnet) [inline cristin-person/src/test/java/no/unit/nva/cristin/person/handler/personAuthorized.feature:35]: Have you tried without token at all? Sometimes it only gives 4xx when supplied with some token that is invalid meaning not supplying a token at all can give 200

**Review by torarnet — APPROVED** [review]
Approved but should wait for approval by all reviewers before merge since this is important functionality

**Pull request merged**
